src/components/__tests__/Mount.js

```diff
@@ -27,7 +29,9 @@ it('create, navigate, and edit thoughts', async () => {
   // create top subthought
   windowEvent('keydown', { key: 'Enter', shiftKey: true, ctrlKey: true })
 
-  jest.runAllTimers()
+  act(() => {
+    jest.runOnlyPendingTimers()
+  })
```

Comment:
> Could you explain what `act` does here? I've never quite understood it.

`act` makes sure all state updates have been processed and applied to the DOM before assertion is made. Generally all the `api` provided by enzyme is already wrapped in `act`. But timeout functions happens outside the React callstack. So we need to wrap it with `act` our self when using jest fake timers. So basically `act` is being used here to make sure the state updates happening inside those timeout functions are processed before assertion is made.

You check out the docs for more details.
https://reactjs.org/docs/testing-recipes.html#act

Here is a great article on when to use `act`
https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#1-when-using-jestusefaketimers

Reply:
> Also, what's the significance of using `runOnlyPendingTimers` vs `runAllTimers`?

It is used to prevent endless loop caused by recursive timer functions. I was getting these issues. 
https://jestjs.io/docs/en/timer-mocks#run-pending-timers

Reply:
> > Could you explain what `act` does here? I've never quite understood it.
> 
> `act` makes sure all state updates have been processed and applied to the DOM before assertion is made. Generally all the `api` provided by enzyme is already wrapped in `act`. But timeout functions happens outside the React callstack. So we need to wrap it with `act` our self when using jest fake timers. So basically `act` is being used here to make sure the state updates happening inside those timeout functions are processed before assertion is made.
> 
> You check out the docs for more details.
> [reactjs.org/docs/testing-recipes.html#act](https://reactjs.org/docs/testing-recipes.html#act)
> 
> Here is a great article on when to use `act`
> [kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#1-when-using-jestusefaketimers](https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning#1-when-using-jestusefaketimers)

Awesome, thanks!

> > Also, what's the significance of using `runOnlyPendingTimers` vs `runAllTimers`?
> 
> It is used to prevent endless loop caused by recursive timer functions. I was getting these issues.
> [jestjs.io/docs/en/timer-mocks#run-pending-timers](https://jestjs.io/docs/en/timer-mocks#run-pending-timers)

Interesting! That makes sense. I hadn't run into that yet.